sled-hardware/src/lib.rs

```diff
@@ -286,17 +289,69 @@ pub struct Disk {
 }
 
 impl Disk {
+    /// Parses a disk, ensuring that all expected partitions exist.
     #[allow(dead_code)]
     pub fn new(
         log: &Logger,
         unparsed_disk: UnparsedDisk,
     ) -> Result<Self, DiskError> {
         let paths = &unparsed_disk.paths;
+        let variant = unparsed_disk.variant;
         // First, ensure the GPT has the right format. This does not necessarily
         // mean that the partitions are populated with the data we need.
-        let partitions =
-            ensure_partition_layout(&log, &paths, unparsed_disk.variant)?;
+        let partitions = ensure_partition_layout(&log, &paths, variant)?;
 
+        Self::initialize_swap(&log, &paths, &partitions)?;
+        let zpool_name =
+            Self::initialize_zpool(&log, variant, &paths, &partitions)?;
+
+        Ok(Self {
+            paths: unparsed_disk.paths,
+            slot: unparsed_disk.slot,
+            variant: unparsed_disk.variant,
+            identity: unparsed_disk.identity,
+            partitions,
+            zpool_name,
+        })
+    }
+
+    // Initize a swap device within the DumpDevice partition, if one exists.
+    fn initialize_swap(
+        log: &Logger,
+        paths: &DiskPaths,
+        partitions: &Vec<Partition>,
+    ) -> Result<(), DiskError> {
+        match paths.partition_device_path(
+            &partitions,
+            Partition::DumpDevice,
+            false,
```

Comment:
Part of this is probably my own unfamiliarity with the sled-hardware code, but it might be good to explain why this is going in the DumpDevice partition?

Reply:
I've struggled to find a great summary for this in the illumos docs, but I'm really just trying to follow the convention for this which has been set there.

Reading a combination of 
- https://illumos.org/man/8/dumpadm
- https://www.illumos.org/man/8/swap
- https://github.com/illumos/illumos-gate/blob/master/usr/src/cmd/swap/swap.c

I was able to infer:

- Apparently, it's common to use swap space as a dump device
- This is even the default mode of `swap -a` -- if [a swap device is added, and no dump devices exist, that swap device is used as a dump device](https://github.com/illumos/illumos-gate/blob/cd6904adfb4d5d78d93238b8062f8dd3ef70a103/usr/src/cmd/swap/swap.c#L571-L576).

I added a summary in 4039c7a3fffdba392e576e55493433f68ecbe774

Reply:
thanks, this was helpful.